hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/clean/CleanPlanner.java

```diff
@@ -509,7 +509,7 @@ public Pair<Boolean, List<CleanFileInfo>> getDeletePaths(String partitionPath) {
    */
   public Option<HoodieInstant> getEarliestCommitToRetain() {
     return CleanerUtils.getEarliestCommitToRetain(
-        hoodieTable.getMetaClient().getActiveTimeline().getCommitsTimeline(),
+        hoodieTable.getMetaClient().getActiveTimeline().getCommitsAndMergesTimeline(),
```

Comment:
Previously, there was no consideration of inflight compaction instant. When partitioning by time, if the inflight compaction instant is in the previous partition and is skipped for incremental cleaning, it will cause the log file not be cleaned up.

Reply:
Nice catch ~